account_payment_order/models/account_payment.py

```diff
@@ -47,3 +48,17 @@ def _compute_payment_method_line_fields(self):
                     and pay.available_payment_method_line_ids.code == "manual"
                 )
         return res
+
+    def _prepare_move_line_default_vals(self, write_off_line_vals=None):
+        """Overwrite date_maturity of the move_lines that are generated when related
+        to a payment order and a specific configuration.
+        """
+        vals_list = super()._prepare_move_line_default_vals(
+            write_off_line_vals=write_off_line_vals
+        )
+        if self.payment_order_id and self.payment_line_ids:
+            payment_mode = self.payment_order_id.payment_mode_id
+            if payment_mode.default_transfer_move_date_prefered == "now_date_maturity":
+                for vals in vals_list:
+                    vals["date_maturity"] = self.payment_line_ids[0].ml_maturity_date
```

Comment:
It's the payment date, not the maturity date:

```suggestion
                    vals["date_maturity"] = self.payment_line_ids[0].date
```